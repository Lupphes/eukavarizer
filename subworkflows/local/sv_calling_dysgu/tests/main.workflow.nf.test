nextflow_workflow {

    name "Test Workflow SV_CALLING_DYSGU"
    script "../main.nf"
    workflow "SV_CALLING_DYSGU"
    config "subworkflows/local/sv_calling_dysgu/tests/nextflow.config"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "sv_calling_dysgu"

    test("Illumina BAM input - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', single_end: false, sample: 'sample1', patient: 'patient1'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ])
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.vcf != null },
                { assert workflow.out.vcfgz != null },
                { assert workflow.out.tbi != null },
                { assert workflow.out.csi != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("Multiple samples - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id: 'sample1', single_end: false, sample: 'sample1', patient: 'patient1'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ],
                    [
                        [id: 'sample2', single_end: false, sample: 'sample2', patient: 'patient1'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ]
                )
                input[1] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[2] = Channel.value([[id: 'genome'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.vcf != null },
                { assert workflow.out.vcfgz != null },
                { assert workflow.out.tbi != null },
                { assert workflow.out.csi != null }
            )
        }
    }
}
