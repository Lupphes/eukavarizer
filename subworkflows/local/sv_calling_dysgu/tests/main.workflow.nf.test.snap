{
    "Illumina BAM input - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcfgz": [
                    [
                        {
                            "id": "test-dysgu-svync",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1"
                        },
                        "re-sample1-null-false-dysgu.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-19T02:50:06.28248063"
    }
}